src/coreclr/tools/Common/Compiler/CompilerTypeSystemContext.Validation.cs

```diff
@@ -153,9 +153,16 @@ private static TypeDesc EnsureLoadableTypeUncached(TypeDesc type)
                     {
                         ThrowHelper.ThrowTypeLoadException(ExceptionStringID.ClassLoadGeneral, type);
                     }
+                }
 
-                    // TODO: validate constraints
+                // Don't validate constraints with crossgen2 - the type system is not set up correctly
+                // and doesn't see generic interfaces on arrays.
+#if !READYTORUN
```

Comment:
I peeled this off from #84404. See the failing test leg with crossgen2. This file is shared with crossgen2, but crossgen2 sets up the type system in a way that `object[]` doesn't implement `IEnumerable<object>` so we can't run this logic there:

https://github.com/dotnet/runtime/blob/1487522704c23df9b7aef2a48874c38a1345bfaa/src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilerContext.cs#L134-L141